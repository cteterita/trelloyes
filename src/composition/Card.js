import React from 'react';
import './Card.css';
import { render } from '@testing-library/react';

class Card extends React.Component {
    render() {
        return (
            <div className='Card'>
                <button type="button">delete</button>
                <h3>{this.props.title}</h3>
                <p>{this.props.content}</p>
            </div>
        );
    }
}

export default Card;