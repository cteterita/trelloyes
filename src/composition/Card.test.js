import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card'; 

describe('Card component', () =>{
    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render UI as expected', () => {
        const tree = renderer.create(<Card title="Test1" content="Content1" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});