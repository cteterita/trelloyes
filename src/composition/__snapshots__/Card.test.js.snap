// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render UI as expected 1`] = `
<div
  className="Card"
>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
  <h3>
    Test1
  </h3>
  <p>
    Content1
  </p>
</div>
`;
